static/app/utils/profiling/gl/utils.ts

```diff
@@ -535,3 +554,66 @@ export function computeClampedConfigView(
 
   return new Rect(clampedX, clampedY, clampedWidth, clampedHeight);
 }
+
+function isBetween(num: number, low: number, high: number) {
+  return num >= low && num <= high;
+}
+
+type Bounds = [number, number];
+function computeHighlightedBounds(bounds: Bounds, trim: TrimTextCenter): Bounds {
```

Comment:
Can you add some comments on the general logic flow of this code, maybe it's just me, but I'm not exactly sure I follow how it works

Reply:
😅  yeah for sure i'll add in comments. the gist here is that we compensate for the trimmed text coming in, some conditional logic on whether or not the matched indices fell within the ellipsis followed by the possibility of the indices shifting by `n` because we truncated the text in the middle by `n`

it's not pretty but its effective